chunky/src/java/se/llbit/chunky/model/DecoratedPotModel.java

```diff
@@ -0,0 +1,169 @@
+        new Vector3(5 / 16., 1, 11 / 16.),
+        new Vector3(5 / 16., 20 / 16., 5 / 16.),
+        new Vector4(0 / 32., 6 / 32., 22 / 32., 26 / 32.)),
+      new Quad( // east
+        new Vector3(11 / 16., 1, 11 / 16.),
+        new Vector3(11 / 16., 1, 5 / 16.),
+        new Vector3(11 / 16., 20 / 16., 11 / 16.),
+        new Vector4(12 / 32., 18 / 32., 22 / 32., 26 / 32.)),
+      new Quad( // top
+        new Vector3(5 / 16., 20 / 16., 11 / 16.),
+        new Vector3(11 / 16., 20 / 16., 11 / 16.),
+        new Vector3(5 / 16., 20 / 16., 5 / 16.),
+        new Vector4(6 / 32., 12 / 32., 26 / 32., 32 / 32.)),
+    };
+
+    private final Material material = new TextureMaterial(Texture.decoratedPotBase);
+    private final String facing;
+
+    public DecoratedPotSpoutEntity(Vector3 position, String facing) {
+      super(position);
+      this.facing = facing;
+    }
+
+    @Override
+    public Collection<Primitive> primitives(Vector3 offset) {
+      Collection<Primitive> primitives = new LinkedList<>();
+      Transform transform = Transform.NONE
+        .translate(position.x + offset.x, position.y + offset.y, position.z + offset.z);
+      for (Quad quad : rotateToFacing(facing, QUADS)) {
+        quad.addTriangles(primitives, material, transform);
+      }
+      return primitives;
+    }
+
+    public static Entity fromJson(JsonObject json) {
+      return new DecoratedPotSpoutEntity(
+        JsonUtil.vec3FromJsonObject(json.get("position")),
+        json.get("facing").stringValue("north")
+      );
+    }
+
+    @Override
+    public JsonValue toJson() {
+      JsonObject json = new JsonObject();
+      json.add("kind", "decoratedPotSpout");
+      json.add("position", position.toJson());
+      json.add("facing", facing);
+      return json;
+    }
+  }
+
+  public DecoratedPotModel(String facing, String[] shards) {
+    super(facing, DEFAULT_QUADS, new Texture[]{
+      // shards[0] top crafting slot -> north
+      getTextureForShard(shards[0]),
+      // shards[3] bottom crafting slot -> south
+      getTextureForShard(shards[3]),
+      // shards[1] left crafting slot -> west
+      getTextureForShard(shards[1]),
+      // shards[2] right crafting slot -> east
+      getTextureForShard(shards[2]),
+      Texture.decoratedPotBase, // top
+      Texture.decoratedPotBase  // bottom
+    });
+  }
+
+  private static Texture getTextureForShard(String shard) {
+    if(shard == null)
+      return Texture.decoratedPotSide;
+    switch(shard) {
+      case "minecraft:pottery_shard_skull":
+        return Texture.decoratedPotPatternSkull;
+      case "minecraft:pottery_shard_prize":
+        return Texture.decoratedPotPatternPrize;
+      case "minecraft:pottery_shard_archer":
+        return Texture.decoratedPotPatternArcher;
+      case "minecraft:pottery_shard_arms_up":
+        return Texture.decoratedPotPatternArmsUp;
+      default:
+        throw new IllegalArgumentException("Unknown pottery shard: " + shard);
```

Comment:
For better resilience, we could instead lof a warning and use `Texture.decoratedPotSide`.